lib/logstash/plugin_mixins/opensearch/api_configs.rb

```diff
@@ -20,6 +20,23 @@ module APIConfigs
         # Password to authenticate to a secure OpenSearch cluster
         :password => { :validate => :password },
 
+        # Credential resolution logic works as follows:
+        #
+        # - User passed aws_access_key_id and aws_secret_access_key in opensearch configuration
+        # - Environment Variables - AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+        #   (RECOMMENDED since they are recognized by all the AWS SDKs and CLI except for .NET),
+        #   or AWS_ACCESS_KEY and AWS_SECRET_KEY (only recognized by Java SDK)
+        # - Credential profiles file at the default location (~/.aws/credentials) shared by all AWS SDKs and the AWS CLI
+        # - Instance profile credentials delivered through the Amazon EC2 metadata service
```

Comment:
Why is this comment here? It seems kind of random.

Reply:
I just tried to paste it from AES as it is still related to auth_type. If you feel it is not required, I will remove it.

Reply:
i will move this close to aws_access_key_id/ aws_secret_access_key config

Reply:
In api_configs.rb, we will not declare aws_access_key_id and aws_secret_access_key. We will only define auth_type as a dictionary and whatever is defined inside auth_type is passed onto options and later validated in manticore_adapter.rb

Reply:
I would add if auth type is aws_iam then following credentials are expected to line 23

Reply:
updated